### PR TITLE
sql,ccl: Changed Error code for experimental features

### LINKED ISSUE #84552: sql: Different error code for experimental/preview features
I realized that we return error code 0A000 for experimental/preview features that require an additional setting.
```
demo@127.0.0.1:26257/movr> alter table vy alter column i set data type int4;
ERROR: ALTER COLUMN TYPE from int to int4 is only supported experimentally
SQLSTATE: 0A000
```

**Desired behavior**
Return a different error code for preview features so that 0A000 truly differentiates unsupported syntax

Epic CRDB-14049
Jira issue: CRDB-17729

### PR DESCRIPTION
**Pull request by e-mbrown**
Resolves #84552 

Defined a new cockroach extension error code to distinguish
experimental features from unsupported features.

Release note: None

### CODE REVIEW

File: pkg/sql/pgwire/pgcode/codes.go [hunk 1]
```diff
@@ -386,4 +386,8 @@ var (
 	// UnsatisfiableBoundedStaleness signals that the bounded staleness query
 	// cannot be satisfied.
 	UnsatisfiableBoundedStaleness = MakeCode("XCUBS")
+
+	// ExperimentalFeature signals that a feature we supported experimentally is being
+	//used without the session variable being enabled.
```
Comment (rafiss) [inline pkg/sql/pgwire/pgcode/codes.go:391]: nit: add a space after `//`

**Review by rafiss — APPROVED** [review]
small nit, then lgtm

out of curiosity, how did you find all the places to change? just wondering if it's possible there are more

**Review by e-mbrown — COMMENTED** [review]
I found all usages of `pgcode.NotSupported` and combed through the results. I could do another round just in case.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84847#-:-N7WsRZTEyT_rwWqipQA:b-f5n1iv)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

**Review by rafiss — APPROVED** [review]

**Comment by e-mbrown** [comment]
TFTR 

bors r=rafiss

**Comment by rafiss** [comment]
blathers backport 22.1 21.2

**Pull request merged**